test/metrics/metrics.jl

```diff
@@ -0,0 +1,215 @@
+using ADRIA: metrics
+using ADRIA: DataFrame, DataFrameRow, names
+using ADRIA: YAXArray, DataCube, ZeroDataCube
+
+if !@isdefined(TEST_RS)
+    const TEST_DOM, TEST_N_SAMPLES, TEST_SCENS, TEST_RS = test_rs()
+end
+
+function _test_metrics(
+    metric::metrics.Metric, params::Tuple; relative::Bool=false
+)::Nothing
+    metric_result = metric(params...)
+    @test all(haskey.([metric_result.properties], (:metric_name, :metric_unit)))
+    @test all(0.0 .<= metric_result.data)
+    relative && @test all(metric_result.data .<= 1.0)
+    return nothing
+end
+
+function test_relative_metric(metric::metrics.Metric, params::Tuple)::Nothing
+    return _test_metrics(metric, params; relative=true)
+end
+
+function test_absolute_metric(metric::metrics.Metric, params::Tuple)::Nothing
+    return _test_metrics(metric, params; relative=false)
+end
+
+@testset "metrics.jl" begin
+    n_timesteps::Int64 = length(TEST_RS.env_layer_md.timeframe)
+    n_groups::Int64 = length(ADRIA.coral_spec().taxa_names)
+    n_sizes::Int64 = length(ADRIA.coral_spec().params.name) / n_groups
+    n_group_sizes::Int64 = n_groups * n_sizes
+    n_locations::Int64 = length(TEST_RS.coral_dhw_tol_log.locations)
+    n_scenarios::Int64 = size(TEST_SCENS, 1)
+    test_scens_datacube::YAXArray{Float64,2} = DataCube(
+        Matrix(TEST_SCENS); scenarios=1:n_scenarios, factors=names(TEST_SCENS)
+    )
+    cover_params = (
+        n_timesteps=n_timesteps,
+        n_group_sizes=n_group_sizes,
+        n_locations=n_locations,
+        n_scenarios=n_scenarios
+    )
+
+    coral_cover::YAXArray{Float64,4} = Factories.coral_cover(; cover_params...)
+    zero_coral_cover::YAXArray{Float64,4} = Factories.zero_coral_cover(; cover_params...)
+    full_coral_cover::YAXArray{Float64,4} = Factories.full_coral_cover(; cover_params...)
+    test_covers::Vector{YAXArray{Float64,4}} = [
+        coral_cover, zero_coral_cover, full_coral_cover
+    ]
+    k_area::Vector{Float64} = round.(rand(n_locations) .* 100; digits=2)
+
+    @testset "relative_cover" begin
+        test_relative_metric(metrics.relative_cover, (TEST_RS,))
+        for cover in test_covers
+            test_relative_metric(metrics.relative_cover, (cover,))
+        end
+    end
```

Comment:
Should we split this file according to the metric types so the tests are more organised?

e.g., a file for the various kinds of `cover`, another for `shelter`, another for `biodiversity` (which is currently just evenness), etc?

Reply:
In my opinion we should reorganize the metrics module and then reorganize this to reflect that. The way it is now it seems easier for me to find a particular test I'm looking for if it is organized by file. The problem, again, is that the file organization is not so good right now.

Reply:
You could equally organise the tests and change the metric module to reflect the tests.

> The way it is now it seems easier for me to find a particular test I'm looking for if it is organized by file.

I'm not sure I follow. The top-level name is `metrics.jl` which the corresponding file, as you say (and I agree) is not organised well, so I'm not sure how it helps?

I `ctrl+shift+f` and search for the specific metric.